CHANGELOG.md

```diff
@@ -1,20 +1,28 @@
 # Release History
 
-## 1.17.0 (TBD)
+## 1.17.0 (2024-05-21)
 
-### New Features
+### Snowpark Python API Updates
+
+#### New Features
```

Comment:
Can we merge the two "New Features" sections?

Reply:
This new features is only used for python API.

Reply:
Oh, that makes sense. Sorry didn't notice that these headers are nested separately under `Snowpark Python API Updates` and `Snowpark pandas API Updates`. Ok, great!